SensusService/Probes/User/ScriptRunner.cs

```diff
@@ -492,11 +517,11 @@ private void StartRandomTriggerCallbacksAsync()
                         DateTime triggerWindowEnd = default(DateTime);
                         DateTime now = DateTime.Now;
                         bool foundTriggerWindow = false;
-                        foreach (Tuple<int, int> randomTriggerWindow in _randomTriggerWindows)
-                            if (randomTriggerWindow.Item1 > now.Hour)
+                        foreach (Tuple<DateTime, DateTime> randomTriggerWindow in _randomTriggerWindows)
+                            if (randomTriggerWindow.Item1 > now)
```

Comment:
I think the following changes will solve this issue without the need to change the comparisons. I've tested it by entering dates in the past and it seems to work.

                        // find next future trigger window
                        DateTime triggerWindowStart = default(DateTime);
                        DateTime triggerWindowEnd = default(DateTime);
                        DateTime now = DateTime.Now;
                        bool foundTriggerWindow = false;
                        foreach (Tuple<DateTime, DateTime> randomTriggerWindow in _randomTriggerWindows)
                            if (randomTriggerWindow.Item1.Hour > now.Hour || (randomTriggerWindow.Item1.Hour == now.Hour && randomTriggerWindow.Item1.Minute > now.Minute))
                            {
                                triggerWindowStart = new DateTime(now.Year, now.Month, now.Day, randomTriggerWindow.Item1.Hour, randomTriggerWindow.Item1.Minute, 0);
                                triggerWindowEnd = new DateTime(now.Year, now.Month, now.Day, randomTriggerWindow.Item2.Hour, randomTriggerWindow.Item2.Minute, 0);
                                foundTriggerWindow = true;
                                break;
                            }

                        // if there were no future trigger windows, skip to the next day and use the first trigger window
                        if (!foundTriggerWindow)
                        {
                            Tuple<DateTime, DateTime> firstRandomTriggerWindow = _randomTriggerWindows.First();
                            triggerWindowStart = new DateTime(now.Year, now.Month, now.Day, firstRandomTriggerWindow.Item1.Hour, firstRandomTriggerWindow.Item1.Minute, 0).AddDays(1);
                            triggerWindowEnd = new DateTime(now.Year, now.Month, now.Day, firstRandomTriggerWindow.Item2.Hour, firstRandomTriggerWindow.Item2.Minute, 0).AddDays(1);
                        }